### ISSUE TITLE
Can't set a breakpoint

### ISSUE THREAD

**Issue by SamyTahar** [jupyterlab/debugger#389]
Hi Team,

I wanted to try your plugin for Jupyter Lab unfortunately when I try to put a breakpoint on a line the point is not set on the click it seems to be a bug.

If not what would be the issue?  

Thanks for your help,
Samy

**Comment by SylvainCorlay** [comment]
Did you use the xpython kernel and enabled the debugger.

**Comment by SamyTahar** [comment]
Yes, I did use xpython and enable the debugger. I can see the orange dot when the cursor is over the line numbers but I click nothing happen the breakpoint is not set.

**Comment by KsavinN** [comment]
@SamyTahar  Could you send screenshot or gif? That would help us to resolve your issue :)

**Comment by SamyTahar** [comment]
Here it is
![2020-03-11 13 14 01](https://user-images.githubusercontent.com/6088286/76415973-7eb31380-639a-11ea-8f57-188a1d97cc11.gif)

**Comment by martinRenou** [comment]
Do you get an error in the JavaScript console? `ctrl + shift + J` on Chrome, `ctrl + shift + K` on Firefox

**Comment by SamyTahar** [comment]
When I refresh the page and reload the xpython kernel I have this error
![image](https://user-images.githubusercontent.com/6088286/76417746-b5d6f400-639d-11ea-9072-4bda2760acc7.png)

**Comment by KsavinN** [comment]
@SamyTahar Do you get in the console some error after a click, when you want to set a breakpoint?

**Comment by SamyTahar** [comment]
No error when I click ...

**Comment by SamyTahar** [comment]
I did install everything again on anaconda base env and it seems to work however when I did create my own env and install all the requirements I had the issue with the breakpoint.

**Comment by jtpio** [comment]
@SamyTahar do you think you could summarize all the steps that led to the issue with the breakpoints?

Thanks!

**Comment by SamyTahar** [comment]
@jtpio 
- So I had anaconda installed in the first place. 
- I have created a conda env named pytorch and I had install pytorch jupyter lab and all dependencies then the debugger.

After that, I used the anaconda env (base) that comes with anaconda and I installed all dependencies again pytorch jupyter lab then the debugger

**Comment by daviddemeij** [comment]
I am experiencing exactly the same issue. I also installed the debugger in an existing conda environment.  

I have the following Jupyter packages installed in conda:
```
jupyter                   1.0.0                      py_2    conda-forge
jupyter-contrib-core      0.3.3                    pypi_0    pypi
jupyter-contrib-nbextensions 0.5.1                    pypi_0    pypi
jupyter-highlight-selected-word 0.2.0                    pypi_0    pypi
jupyter-latex-envs        1.4.6                    pypi_0    pypi
jupyter-nbextensions-configurator 0.4.1                    pypi_0    pypi
jupyter_client            5.3.4                    py36_1    conda-forge
jupyter_console           5.1.0                    py36_0    conda-forge
jupyter_contrib_core      0.3.3                      py_2    conda-forge
jupyter_contrib_nbextensions 0.5.1                    py36_0    conda-forge
jupyter_core              4.6.1                    py36_0    conda-forge
jupyter_highlight_selected_word 0.2.0                 py36_1000    conda-forge
jupyter_latex_envs        1.4.4                 py36_1000    conda-forge
jupyter_nbextensions_configurator 0.4.1                    py36_0    conda-forge
jupyterlab                1.2.6                      py_0    conda-forge
jupyterlab_server         1.0.6                      py_0    conda-forge
```
And I only have two extensions installed:
```
@jupyterlab/toc
@jupyterlab/debugger
```

**Comment by martinRenou** [comment]
> I also have some error when using xpython while trying to import certain packages, but this is probably not related.

Which package were you trying to import? Were you using Python magics in the cell? Python magics do not work yet in xpython

**Comment by jtpio** [comment]
This sounds very similar to https://github.com/jupyterlab/debugger/issues/319.

@daviddemeij  would it be possible to test with Python 3.7?

**Comment by daviddemeij** [comment]
> > I also have some error when using xpython while trying to import certain packages, but this is probably not related.
> 
> Which package were you trying to import? Were you using Python magics in the cell? Python magics do not work yet in xpython

Yes, I was actually using line magic and when I removed it I didn't get this error anymore. And for some reason I can also set breakpoints now, but not sure if this is because of removing the line magic or because I restarted the kernel again. So everything seems to work now, thanks!

**Comment by afshin** [comment]
Thanks for the bug report. Since this extension is still in active development heading toward a 1.0 release later this year, we are not fixing bugs on the previous 0.2 branch.

Are you able to upgrade to JupyterLab 2.x and install the `@jupyterlab/debugger` v0.2.x?

**Comment by emirceyani** [comment]
Hello, I also couldn't set the breakpoint like @SamyTahar .  I first installed with pip but didn't give an inch. Now, reinstalled the xeus-python with 
`conda install -c conda-forge xeus-python ` which installs version 0.7.1 

Maybe you can check xeus-python version.

Hope this helps

**Comment by derkuci** [comment]
I had the same issue.  I noticed that the ptvsd package complained about 'sys' has no attribute 'argv'.  So I typed in Jupyterlab "sys.argv = []", and then I can set breakpoints in that session.

The installation was via conda.  xeus-python 0.6.11 and ptvsd 4.3.2.

**Comment by amapic** [comment]
I also had the same issue with python 3.6.8. After installing jupyterlab2.x and @jupyterlab/debugger v0.2.x, it was not working either. I created a new env with python 3.7 and then it was ok.

**Comment by pratikchhapolika** [comment]
> Hi Team,
> 
> I wanted to try your plugin for Jupyter Lab unfortunately when I try to put a breakpoint on a line the point is not set on the click it seems to be a bug.
> 
> If not what would be the issue?
> 
> Thanks for your help,
> Samy

Same ssue with me on Windows 10. :(

**Comment by rohan-paul** [comment]
Same issue in Ubuntu 20.04

**Comment by Gusilin** [comment]
having the same issue, but got this error anyone can help me ples?? Thanks
![image](https://user-images.githubusercontent.com/68452531/98756921-dd091980-23a1-11eb-826a-670e01affafe.png)

**Comment by mahmud97** [comment]
While I try to set breakpoints this error appears
![Error](https://user-images.githubusercontent.com/15126836/100832853-e4987b80-3492-11eb-8569-68817c725f6f.png)

**Comment by mahmud97** [comment]
> Hello, I also couldn't set the breakpoint like @SamyTahar . I first installed with pip but didn't give an inch. Now, reinstalled the xeus-python with
> `conda install -c conda-forge xeus-python ` which installs version 0.7.1
> 
> Maybe you can check xeus-python version.
> 
> Hope this helps

I installed using 
1.jupyter labextension install @jupyterlab/debugger
2.jupyter labextension install @jupyterlab/debugger
but still encountering the error while trying to set breakpoints

**Comment by JohanMabille** [comment]
Can you check with last `xeus-python` (should be 0.9.1). There has been a recent fix in the way breakpoints are sent to and retrieve from the kernel. Although this cannot explain the original issue, this may be the cause of the more recent similar problems.

Also the kernel switched from ptvsd to debugpy, which fixes a lot of issues we had with ptvsd (the `sys.argv` among them).

**Comment by fvollmer** [comment]
I'm experiencing the same issue with xeus-python 0.9.1

(project) C:\Users\user\project>conda list xeus-python
# packages in environment at C:\Users\user\Anaconda3\envs\project
#
# Name                    Version                   Build  Channel
xeus-python               0.9.1            py37h7f3b3cb_0    conda-forge

**Comment by JohanMabille** [comment]
Can you precise the context:

- Operating System (and version)
- Python version
- Browser and version

?

We are also tracking this issue [here](https://github.com/jupyterlab/jupyterlab/issues/9414).

**Comment by fvollmer** [comment]
Sure:

Windows 10, 20H2, Build 19042.630
Python 3.7.6
Firefox 83.0 (64-Bit), Chrome 87.0.4280.88

![grafik](https://user-images.githubusercontent.com/16699443/101399395-b3271200-38cf-11eb-83d7-c2f07df8768d.png)
![grafik](https://user-images.githubusercontent.com/16699443/101399505-dce03900-38cf-11eb-9d89-5371c07cc5ef.png)

**Comment by JohanMabille** [comment]
We've released xeus-python 0.9.3, it should fix this issue. See https://github.com/jupyterlab/jupyterlab/issues/9414

**Comment by jtpio** [comment]
Thanks @JohanMabille!

Closing as fixed. @fvollmer feel free to comment in https://github.com/jupyterlab/jupyterlab/issues/9414 if you are still experiencing the issue. Or open a new issue in the JupyterLab repo directly. Thanks!
